docs/current/sdk/go/guides/759201-gitlab-google-cloud.md

```diff
@@ -0,0 +1,270 @@
+    - Once the application is built, it moves to the second stage of the build. It again uses the client's `Container().From()` method to initialize a new container from an `alpine` base image.
+    - It uses the previous `Container` object's `WithFile()` method to transfer the compiled binary file from the first stage to the new container filesystem.
+    - It sets the container entrypoint to the binary file using the `withEntrypoint()` method.
+    - It uses the container object's `Publish()` method to publish the container to Google Container Registry, and prints the SHA identifier of the published image.
+    - It creates a Google Cloud Run client, creates a service request instructing the Google Cloud Run service to use the newly-published container image, and sends the requests to the Google Cloud Run API.
+
+  :::tip
+  Most `Container` object methods return a revised `Container` object representing the new state of the container. This makes it easy to chain methods together. Dagger evaluates pipelines "lazily", so the chained operations are only executed when required - in this case, when the `Publish()` method is called.
+  :::
+
+1. Run the following command to update `go.sum`:
+
+  ```shell
+  go mod tidy
+  ```
+
+## Step 4: Test the Dagger pipeline on the local host
+
+Test the Dagger pipeline as follows:
+
+{@include: ../../../partials/_google-cloud-sdk-credentials-setup.md}
+
+3. Run the pipeline:
+
+  ```shell
+  go run ci/main.go
+  ```
+
+Dagger performs the operations defined in the pipeline script, logging each operation to the console. At the end of the process, the built container is deployed to Google Cloud Run and a message similar to the one below appears in the console output:
+
+  ```shell
+  Deployment for image gcr.io/PROJECT/myapp@sha256:b1cf... now available at https://...run.app
+  ```
+
+Browse to the URL shown in the deployment message to see the running application.
+
+If you deployed the example application from [Appendix A](#appendix-a-create-a-gitlab-repository-with-an-example-go-application), you see the output below:
+
+```shell
+Hello, Dagger!
+```
+
+## Step 5: Create a GitLab CI/CD pipeline
+
+Dagger executes your pipelines entirely as standard OCI containers. This means that the same pipeline will run the same, whether on on your local machine or a remote server.
+
+This also means that it's very easy to move your Dagger pipeline from your local host to GitLab - all that's needed is to transfer the pipeline script from your local clone to your GitLab repository, and then define a GitLab CI/CD pipeline to run it on every commit.
+
+1. Create a new GitLab CI/CD pipeline configuration file in your application directory at `.gitlab-ci.yml` with the following content:
+
+  ```yaml file=../snippets/gitlab-google-cloud/gitlab-ci.yml
```

Comment:
I have tested this with a self-hosted runner. However I don't have access to GitLab's shared runners so need some help here. Can someone with access to shared runners please check if this configuration works as advertised?

Reply:
Tested on a shared runner, works great! 🚀